metadata/metadata_service/proxy/atlas_proxy.py

```diff
@@ -187,22 +220,27 @@ def _create_bookmark(self, entity: AtlasEntityWithExtInfo, user_guid: str, bookm
         bookmark_entity = type_coerce(bookmark_entity, AtlasEntityWithExtInfo)
         self.client.entity.create_entity(bookmark_entity)
 
-    def _get_bookmark_entity(self, entity_uri: str, user_id: str) -> AtlasEntityWithExtInfo:
+    def _get_bookmark_entity(self, entity_uri: str, user_id: str,
+                             resource_type: ResourceType) -> AtlasEntityWithExtInfo:
         """
-        Fetch a Bookmark entity from parsing table uri and user id.
+        Fetch a Bookmark entity from parsing entity uri and user id.
         If Bookmark is not present, create one for the user.
         :param entity_uri:
         :param user_id: Qualified Name of a user
         :return:
         """
-        table_info = AtlasTableKey(entity_uri).get_details()
+        if resource_type == ResourceType.Table:
+            entity_info = AtlasTableKey(entity_uri).get_details()
 
-        schema = table_info.get('schema')
-        table = table_info.get('table')
-        database = table_info.get('database')
-        cluster = table_info.get('cluster')
+            schema = entity_info.get('schema')
+            table = entity_info.get('table')
+            database = entity_info.get('database')
+            cluster = entity_info.get('cluster')
 
-        bookmark_qn = f'{schema}.{table}.{database}.{user_id}.bookmark@{cluster}'
+            bookmark_qn = f'{schema}.{table}.{database}.{user_id}.bookmark@{cluster}'
+
+        else:
+            bookmark_qn = f'{entity_uri}/{resource_type.name.lower()}/{user_id}/bookmark'
```

Comment:
can we have it `{entity_uri}/{resource_type.name.lower()}/bookmark/{user_id}` ?

Reply:
changed